src/ui/public/vislib/components/color/heatmap_color.js

```diff
@@ -0,0 +1,69 @@
+import _ from 'lodash';
+import colormaps from './colormaps';
+
+function enforceBounds(x) {
+  if (x < 0) {
+    return 0;
+  } else if (x > 1) {
+    return 1;
+  } else {
+    return x;
+  }
+}
+
+function interpolateLinearly(x, values) {
+  // Split values into four lists
+  const xValues = [];
+  const rValues = [];
+  const gValues = [];
+  const bValues = [];
+  values.forEach(value => {
+    xValues.push(value[0]);
+    rValues.push(value[1][0]);
+    gValues.push(value[1][1]);
+    bValues.push(value[1][2]);
+  });
+
+  let i = 1;
+  while (xValues[i] < x) i++;
+  const width = Math.abs(xValues[i - 1] - xValues[i]);
+  const scalingFactor = (x - xValues[i - 1]) / width;
+  // Get the new color values though interpolation
+  const r = rValues[i - 1] + scalingFactor * (rValues[i] - rValues[i - 1]);
+  const g = gValues[i - 1] + scalingFactor * (gValues[i] - gValues[i - 1]);
+  const b = bValues[i - 1] + scalingFactor * (bValues[i] - bValues[i - 1]);
+  return [enforceBounds(r), enforceBounds(g), enforceBounds(b)];
+}
+
+function getColor(value, colorSchemaName) {
+  if (!_.isNumber(value) || value < 0 || value > 1) {
+    throw new Error('heatmap_color expects a number from 0 to 1 as first parameter');
+  }
+
+  const colorSchema = colormaps[colorSchemaName];
+  if (!colorSchema) {
+    throw new Error('invalid colorSchemaName provided');
+  }
+
+  const color = interpolateLinearly(value, colorSchema);
+  const r = Math.round(255 * color[0]);
+  const g = Math.round(255 * color[1]);
+  const b = Math.round(255 * color[2]);
+  return `rgb(${r},${g},${b})`;
+}
+
+function drawColormap(colorSchema, width = 100, height = 10) {
```

Comment:
i added this as i wanted to add it to dropdown .... however i would need to change that dropdown to div/js based one to add images inside ... would be a nice feature imo .... will check with CJ what he thinks about adding another component